### PR TITLE
nixos/cmt: Update error message when enabling cmt and plasma5

### LINKED ISSUE #69282: Enabling both module cmt.nix and plasma5.nix throws a failed assertion and build aborts
**Describe the bug**
In configuration.nix, if you enable both xserver.services.desktopManager.plasma5 and xserver.services.cmt, assertion fails.

`assertion = !config.services.xserver.libinput.enable;`

Failed assertions:
- cmt and libinput are incompatible, you cannot enable both (in services.xserver).

**To Reproduce**
1. Enable both desktopManager.plasma5.enable and cmt.enable

**Expected behavior**
Failed assertion should happen

**Additional context**
Build with xserver.services.desktopManager.xfce works fine.

**Metadata**
``` 
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.75, NixOS, 20.03.git.6262e83 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information: kcalvinalvin

a list of nixpkgs attributes affected by the problem
`attribute: xserver.services.desktopManager.plasma5, xserver.services.cmt` 

a list of nixos modules affected by the problem
`module: plasma5.nix, cmt.nix`

### PR DESCRIPTION
**Pull request by kcalvinalvin**
Updated error message for the assertion failure when enabling both cmt and plasma5. Fixes https://github.com/NixOS/nixpkgs/issues/69282

###### Motivation for this change
Clearer error message for the user

###### Things done
Updated the error message on assertion failure when enabling libinput in cmt.nix

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: nixos/modules/services/x11/hardware/cmt.nix [hunk 1]
```diff
@@ -30,6 +30,7 @@ in {
 
   config = mkIf cfg.enable {
 
+    services.xserver.libinput.enable = false; #plasma5.nix enables libinput by default which is incompatible with cmt.nix
```
Comment (worldofpeace) [inline nixos/modules/services/x11/hardware/cmt.nix]: We have this assertion below
`!config.services.xserver.libinput.enable`
which produces a message
```
cmt and libinput are incompatible, you cannot enable both (in services.xserver).
```

I think the actual issue is the module doesn't prompt the user to do any proactive change.
Can we change the message so it tells the users to disable `services.xserver.libinput.enable`?

Comment (kcalvinalvin) [reply nixos/modules/services/x11/hardware/cmt.nix]: Sure that could also work.

**Comment by kcalvinalvin** [comment]
Updated the PR to include clearer error message rather than disabling libinput in cmt.nix

File: nixos/modules/services/x11/hardware/cmt.nix [hunk 2]
```diff
@@ -47,7 +47,11 @@ in {
     assertions = [
       {
         assertion = !config.services.xserver.libinput.enable;
-        message = "cmt and libinput are incompatible, you cannot enable both (in services.xserver).";
+        message = ''
+          cmt and libinput are incompatible, you cannot enable both (in services.xserver).
+          plasma5 enables libinput by default. You must declare "services.xserver.libinput.enable = false"
```
Comment (worldofpeace) [inline nixos/modules/services/x11/hardware/cmt.nix]: I think the message about plasma5 is irrelevant, many other desktop manager modules already enable libinput by default. How about
```
cmt and libinput are incompatible, meaning you cannot enable them both.
To use cmt you need to disable libinput with `services.xserver.libinput.enable = false`.
```

Comment (kcalvinalvin) [reply nixos/modules/services/x11/hardware/cmt.nix]: Wouldn’t it be nice to tell the user that libinput is enabled by default somewhere? I was quite perplexed and thought it was a weird bug.

So how about
```
cmt and libinput are incompatible, meaning you cannot enable them both.
To use cmt you need to disable libinput with `services.xserver.libinput.enable = false`. If you haven’t enabled it in configuration.nix, it’s enabled by default on a 
different xserver module.
```

Comment (worldofpeace) [reply nixos/modules/services/x11/hardware/cmt.nix]: I suppose that's fine too :+1:

**Pull request closed without merging**